Fatma- Homework React - week1

FatmaHameed opened the pull request:
I finished all the code. But I tried to deploy it in Netlify, it gave me each time broken deploy site which shows 404 error, page not found, despite that I found all my files when I downloaded them from Netlify as zip. files. I did not know how to solve it.

FatmaHameed commented:
The project is deployed on this [link](https://5fb159ccf58291ac2479952f--reactweek1.netlify.app/)

The exercises are in this [link](https://5fb15bd6cf2b1c8fbf913fe4--reactweek1.netlify.app/)

karamad10 commented on week1/project/hackyourweather/src/App.js:
```diff
@@ -0,0 +1,16 @@
+import './App.css';
+import WeatherBox from './components/weatherBoxes';
+
+const data = require('./city-weather.json');
```
data is very general, this would be more clear if you name it something like `cityWeatherData`

FatmaHameed replied:
Thank you for your comments and feedback. I have already fixed it

karamad10 commented on week1/project/hackyourweather/src/components/weatherBoxes.js:
```diff
@@ -0,0 +1,38 @@
+
+const WeatherBox = () => {
+
+ const weatherList = data.map((city) => {
```
good job on using the map directly in the render and destructuring everything!

karamad10 commented on week1/project/hackyourweather/src/components/weatherBoxes.js:
```diff
@@ -0,0 +1,38 @@
+
+ const weatherList = data.map((city) => {
+   const{name, sys: {country}, weather: [{main}], weather: [{description}], main: {temp_min}, main: {temp_max}, coord: {lat}, coord: {lon}} = city;
+
```
nice destructuring work!

FatmaHameed replied:
Thank you

karamad10 commented on week1/project/hackyourweather/src/components/weatherBoxes.js:
```diff
@@ -0,0 +1,38 @@
+
+    return (
+    <div className="box">
+      <h2>{name}, {country}</h2>
```
semantically this should be an `H1`

FatmaHameed replied:
Thank you for the feedback. I did it as h2 because I had already used h1 for the heading of the project.

karamad10 commented on week1/project/hackyourweather/src/components/weatherBoxes.js:
```diff
@@ -0,0 +1,38 @@
+    <label>max temp: </label><span>{temp_max}</span>
+    </div>
+    <div>
+    <label>Location: </label><span>{lat}, {lon}</span>
```
the `<label>` tag is used to define a label for buttons and inputs and selects etc for accessibility reasons and such, these can all be `<span>`

FatmaHameed replied:
Fixed! Thank you

karamad10 commented on week1/project/hackyourweather/src/components/weatherBoxes.js:
```diff
@@ -0,0 +1,38 @@
+
+const WeatherBox = () => {
+
+ const weatherList = data.map((city) => {
```
every element in a map needs a unique `key` attribute, you should also get an error in the console for that
Do not use the index for it : )

FatmaHameed replied:
Fixed! Thank you

karamad10 commented on week1/react-exercises/src/App.js:
```diff
@@ -0,0 +1,24 @@
+import './App.css';
+import HobbyList from './components/extremeHobbies';
+import Guarantee from './components/customerService';
+import Counter from './components/counting';
```
react components should always be in `SnakeCase`

FatmaHameed replied:
Fixed! Thank you

karamad10 commented on week1/react-exercises/src/components/customerService.js:
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+
+function Guarantee(props){
+const { image, title, description } = props
```
👍

FatmaHameed replied:
Thank you

karamad10 commented on week1/react-exercises/src/components/counting.js:
```diff
@@ -0,0 +1,28 @@
+  );
+};
+
+const Button = ({ setCount, count }) => {
```
👍

FatmaHameed replied:
Thank you

karamad10 commented on week1/react-exercises/src/components/extremeHobbies.js:
```diff
@@ -0,0 +1,19 @@
+
+function Hobbies(props) {
+  const hobbies = props.hobbies;
+  const listHobbies = hobbies.map((hobby) => <li>{hobby}</li>)
```
every element in a map needs a unique key attribute, you should also get an error in the console for that
Do not use the index for it : )

FatmaHameed replied:
Fixed! Thank you

karamad10 commented on week1/react-exercises/src/components/extremeHobbies.js:
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+function Hobbies(props) {
```
would be nicer directly from here like you used it in other places : ) 
`({ hobbies })`

FatmaHameed replied:
Done!

karamad10 commented:
hey @FatmaHameed great job!

just a few comments and on top of that, I'd like to mention the importance of pushing your code properly.
it's a lot to look at when there's a full create-react-app boiler-plate pushed with the home-work at one go
the best thing to do first thing when you create your new react-app is to commit the initial react boiler-plate then start working on your homework
Also very important is properly indenting your code, please take a look at prettier and/or esLint extensions

if you need any help please feel free to reach out : )

FatmaHameed commented:
Hi @karamad10 , thank you for your comments and feedback. I have already addressed your comments and pushed my work again. I also added some styling to the exercises to make it look a bit nicer :) 

Regarding using the prettier and esLint extensions, I have tried to do so, but I do not know why it is not working with me.

Thanks again for reviewing my code :)
